File: BigAnimal/go/delete-cluster.go
// Original work by: EnterpriseDB
// https://www.enterprisedb.com/
// Purpose : Delete BigAnimal clusters
// Author  : Doug Ortiz
// Date    : January 28, 2022
// Version : 1.0
// Copyright © 2020 EnterpriseDB

package main

import (
  "net/http"
  "fmt"
  )

func main() {
     // Assign variables values
     // Set 'token'
     var token = "<yourtoken>"
     // URL, Cluster ID, and Postgres Cluster ID
     var url = "https://portal.biganimal.com/api/v1/clusters"
     pgId := "<yourbiganimalclusterid>"
     filter := "/" + pgId
     filteredurl := url + filter
     fmt.Println("Filter: ", filteredurl)

     // DELETE API Request
     request, errorHttp := http.NewRequest("DELETE", filteredurl, nil)
     if errorHttp != nil {
       panic(errorHttp)
     }
     request.Header.Set("Accept", "*/*")
     request.Header.Set("Authorization", "Bearer " + token)

     // Perform client call
     client := &http.Client{}
     response, errorHttp := client.Do(request)
     // Check for errors
     if errorHttp != nil {
       panic(errorHttp)
     }

     // Close response
     defer response.Body.Close()

     // Provide status
     fmt.Println("Successfully deleted cluster: " + pgId + " !")
}

File: BigAnimal/python/delete-cluster.py
# Original work by: EnterpriseDB
# https://www.enterprisedb.com/
# Purpose : Delete BigAnimal Clusters
# Author  : Doug Ortiz
# Date    : January 28, 2022
# Version : 1.0
# Copyright © 2020 EnterpriseDB

import requests
from requests.structures import CaseInsensitiveDict
import json

# Assign variables values
# Set 'token'
token = "<yourtoken>"
headers = CaseInsensitiveDict()
headers["Accept"] = "*/*"
headers["Authorization"] = "Bearer " + token

# Postgres Cluster ID, URL, filter and filteredurl
# pgId must be retrieved or known prior to making the API call
pgId = "<yourbiganimalclusterid>"
url = "https://portal.biganimal.com/api/v1/clusters"
filter = "/" + pgId
filteredurl = url+filter
print("Filter: ")
print(filteredurl)

# Perform API Delete Request
try:
    response = requests.delete(filteredurl, headers=headers)
    response.raise_for_status()
    print("")
    print("Cluster: " + pgId + " deleted successfully!")
except requests.exceptions.HTTPError as errHTTP:
    print(errHTTP)
except requests.exceptions.ConnectionError as errorConnection:
    print(errorConnection)
except requests.exceptions.Timeout as errorTimeout:
    print(errorTimeout)
except requests.exceptions.RequestException as errException:
    print(errException)
